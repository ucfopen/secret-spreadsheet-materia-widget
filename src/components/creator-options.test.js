import React from 'react'
import Options from './creator-options'
import renderer from 'react-test-renderer'
import { mount, shallow } from '../enzyme'

const props ={
	qset: {
		dimensions: {
			x: 1,
			y: 1,
		},
		spreadsheet: true,
		left: true,
		header: true,
		randomization: 1,
	},
	useSpreadsheet: jest.fn(),
	useTable: jest.fn(),
	useLeftAlign: jest.fn(),
	useCenterAlign: jest.fn(),
	useHeader: jest.fn(),
	toggleInstruction: jest.fn(),
	useQuestion: jest.fn(),
	resetCheckbox: jest.fn(),
}

const makeEvent = value => {
	return {
		target: {
			value: value,
		}
	}
}

describe('CreatorOptions component', function() {

	beforeEach(() => {
		jest.resetModules()
		jest.resetAllMocks()
	})

	test('CreatorOptions renders basic options bar', () => {
		const tree = renderer.create(<Options {... props}/>).toJSON();
		expect(tree).toMatchSnapshot();
	})

	test('CreatorOptions calls useSpreadsheet onKeyDown', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#spreadsheet').simulate('keyDown', {key: 'Enter'})
		expect(props.useSpreadsheet).toReturnTimes(1)
	})

	test('CreatorOptions does not call useSpreadsheet with wrong key', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#spreadsheet').simulate('keyDown', {key: 'a'})
		expect(props.useSpreadsheet).toReturnTimes(0)
	})

	test('CreatorOptions calls useTable onKeyDown', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#table').simulate('keyDown', {key: 'Enter'})
		expect(props.useTable).toReturnTimes(1)
	})

	test('CreatorOptions does not call useTable with wrong key', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#table').simulate('keyDown', {key: 'a'})
		expect(props.useTable).toReturnTimes(0)
	})

	test('CreatorOptions calls useLeftAlign onKeyDown', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#left').simulate('keyDown', {key: 'Enter'})
		expect(props.useLeftAlign).toReturnTimes(1)
	})

	test('CreatorOptions does not call useLeftAlign with wrong key', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#left').simulate('keyDown', {key: 'a'})
		expect(props.useLeftAlign).toReturnTimes(0)
	})

	test('CreatorOptions calls useCenterAlign onKeyDown', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#center').simulate('keyDown', {key: 'Enter'})
		expect(props.useCenterAlign).toReturnTimes(1)
	})

	test('CreatorOptions does not call useCenterAlign with wrong key', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#center').simulate('keyDown', {key: 'a'})
		expect(props.useCenterAlign).toReturnTimes(0)
	})

	test('CreatorOptions calls useHeader onKeyDown', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#header').simulate('keyDown', {key: 'Enter'})
		expect(props.useHeader).toReturnTimes(1)
	})

	test('CreatorOptions does not call useHeader with wrong key', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#header').simulate('keyDown', {key: 'a'})
		expect(props.useHeader).toReturnTimes(0)
	})

	test('CreatorOptions calls useHeader onKeyDown', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#instructions').simulate('keyDown', {key: 'Enter'})
		expect(props.toggleInstruction).toReturnTimes(1)
	})

	test('CreatorOptions does not call useHeader with wrong key', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#instructions').simulate('keyDown', {key: 'a'})
		expect(props.toggleInstruction).toReturnTimes(0)
	})

	test('CreatorOptions calls useHeader onKeyDown', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#question').simulate('keyDown', {key: 'Enter'})
		expect(props.useQuestion).toReturnTimes(1)
	})

	test('CreatorOptions does not call useHeader with wrong key', () => {
		const component = shallow(<Options {... props}/>)
		component.find('#question').simulate('keyDown', {key: 'a'})
		expect(props.useQuestion).toReturnTimes(0)
	})

	test('CreatorOptions renders branches of classNames', () => {
		const props ={
			qset: {
				spreadsheet: false,
				left: false,
				header: false,
				randomization: 0,
			},
		}

		const component = shallow(<Options {... props}/>)
		expect(component.find('#spreadsheet').prop('className')).toEqual('hoverable')

	})

	test('CreatorOptions with negative randomization', () => {
		const event = makeEvent(-1)

		const component = shallow(<Options {... props} />)
		component.instance().handleRandomizationChange(event);
		expect(props.qset.randomization).toEqual(0)

	})

	test('CreatorOptions with normal randomization with header', () => {
		const event = makeEvent(1)

		const component = shallow(<Options {... props} />)
		component.instance().handleRandomizationChange(event);
		expect(props.qset.randomization).toEqual(0)

	})

	test('CreatorOptions with large randomization', () => {
		const event = makeEvent(999)

		const component = shallow(<Options {... props} />)
		component.instance().handleRandomizationChange(event);
		expect(props.qset.randomization).toEqual(0)

	})

	test('CreatorOptions with float as randomization with header', () => {
		const event = makeEvent(0.5)

		const component = shallow(<Options {... props} />)
		component.instance().handleRandomizationChange(event);
		expect(props.qset.randomization).toEqual(0)

	})

	test('CreatorOptions with normal randomization without header', () => {
		const event = makeEvent(1)
		props.qset.header = false

		const component = shallow(<Options {... props} />)
		component.instance().handleRandomizationChange(event);
		expect(props.qset.randomization).toEqual(1)

	})

	test('CreatorOptions with large randomization without header', () => {
		const event = makeEvent(999)

		const component = shallow(<Options {... props} />)
		component.instance().handleRandomizationChange(event);
		expect(props.qset.randomization).toEqual(1)

	})

	test('CreatorOptions with float as randomization without header', () => {
		const event = makeEvent(1.2)

		const component = shallow(<Options {... props} />)
		component.instance().handleRandomizationChange(event);
		expect(props.qset.randomization).toEqual(1)

	})

})
