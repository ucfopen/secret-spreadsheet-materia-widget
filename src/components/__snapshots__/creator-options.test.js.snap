// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorOptions component CreatorOptions renders basic options bar 1`] = `
<div>
  <div
    className="options-bar"
  >
    <div
      className="options"
    >
      <label>
        OPTIONS
      </label>
    </div>
    <div
      className="style"
    >
      <label>
        <strong>
          Style:
        </strong>
      </label>
      <div
        className="active"
        id="spreadsheet-button"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <svg
          height="20px"
          id="spreadsheet-checkbox-checked"
          viewBox="0 0 28 28"
          width="20px"
        >
           
          <path
            d="M0 0v28h28V0H0zm24 24H4V4h20v20zm-2-13l-2.828-2.828-6.768 6.982-3.576-3.576L6 14.406l6.404 6.406L22 11z"
          />
           
        </svg>
        <label>
          Spreadsheet
        </label>
      </div>
      <div
        className="inactive"
        id="table-button"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <svg
          height="20px"
          id="table-checkbox"
          viewBox="0 0 28 28"
          width="20px"
        >
           
          <path
            d="M0 0v28h28V0H0zm24 24H4V4h20v20z"
          />
           
        </svg>
        <label>
          Table
        </label>
      </div>
    </div>
    <div
      className="text"
    >
      <label>
        <strong>
          Text:
        </strong>
      </label>
      <div
        className="active"
        id="left-button"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <svg
          height="20px"
          preserveAspectRatio="xMidYMid meet"
          version="1.0"
          viewBox="0 0 344 344"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
           
          <g
            fill="#000000"
            stroke="none"
            transform="translate(0,344) scale(0.1,-0.1)"
          >
             
            <path
              d="M130 3095 l0 -145 1590 0 1590 0 0 145 0 145 -1590 0 -1590 0 0 -145z"
            />
             
            <path
              d="M130 2410 l0 -140 1105 0 1105 0 0 140 0 140 -1105 0 -1105 0 0 -140z"
            />
             
            <path
              d="M130 1720 l0 -140 1590 0 1590 0 0 140 0 140 -1590 0 -1590 0 0 -140z"
            />
             
            <path
              d="M130 1030 l0 -140 1105 0 1105 0 0 140 0 140 -1105 0 -1105 0 0 -140z"
            />
             
            <path
              d="M130 345 l0 -145 1590 0 1590 0 0 145 0 145 -1590 0 -1590 0 0 -145z"
            />
             
          </g>
           
        </svg>
      </div>
      <div
        className=""
        id="center-button"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <svg
          height="20px"
          preserveAspectRatio="xMidYMid meet"
          version="1.0"
          viewBox="0 0 344 344"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
           
          <g
            fill="#000000"
            stroke="none"
            transform="translate(0,344) scale(0.1,-0.1)"
          >
             
            <path
              d="M130 3095 l0 -145 1590 0 1590 0 0 145 0 145 -1590 0 -1590 0 0 -145z"
            />
             
            <path
              d="M610 2410 l0 -140 1110 0 1110 0 0 140 0 140 -1110 0 -1110 0 0 -140z"
            />
             
            <path
              d="M130 1720 l0 -140 1590 0 1590 0 0 140 0 140 -1590 0 -1590 0 0 -140z"
            />
             
            <path
              d="M610 1030 l0 -140 1110 0 1110 0 0 140 0 140 -1110 0 -1110 0 0 -140z"
            />
             
            <path
              d="M130 345 l0 -145 1590 0 1590 0 0 145 0 145 -1590 0 -1590 0 0 -145z"
            />
             
          </g>
           
        </svg>
      </div>
    </div>
    <div
      className="header"
    >
      <label>
        <strong>
          Header:
        </strong>
      </label>
      <div
        className="active"
        id="header-button"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <svg
          height="20px"
          id="header-checkbox-checked"
          viewBox="0 0 28 28"
          width="20px"
        >
           
          <path
            d="M0 0v28h28V0H0zm24 24H4V4h20v20zm-2-13l-2.828-2.828-6.768 6.982-3.576-3.576L6 14.406l6.404 6.406L22 11z"
          />
           
        </svg>
        <label>
          First row is a header row
        </label>
      </div>
    </div>
    <div
      className="randomize"
    >
      <label>
        <strong>
          Randomize:
        </strong>
      </label>
      <div>
        <label>
          Randomly hide 
          <input
            defaultValue={1}
            onBlur={[Function]}
            placeholder={0}
            type="number"
          />
           cell(s)
        </label>
      </div>
    </div>
  </div>
</div>
`;
